chain/client/src/view_client.rs

```diff
@@ -1095,16 +1095,30 @@ impl Handler<NetworkViewClientMessages> for ViewClientActor {
             }
             NetworkViewClientMessages::GetChainInfo => match self.chain.head() {
                 Ok(head) => {
-                    let height = self.get_height(&head);
-
-                    NetworkViewClientResponses::ChainInfo {
-                        genesis_id: GenesisId {
-                            chain_id: self.config.chain_id.clone(),
-                            hash: *self.chain.genesis().hash(),
-                        },
-                        height,
-                        tracked_shards: self.config.tracked_shards.clone(),
-                        archival: self.config.archive,
+                    match self.runtime_adapter.num_shards(&head.epoch_id) {
+                        Ok(num_shards) => {
+                            // convert config tracked shards
+                            // runtime will track all shards if config tracked shards is not empty
+                            // https://github.com/near/nearcore/issues/4930
+                            let tracked_shards = if self.config.tracked_shards.is_empty() {
+                                vec![]
+                            } else {
+                                (0..num_shards).collect()
+                            };
+                            NetworkViewClientResponses::ChainInfo {
+                                genesis_id: GenesisId {
+                                    chain_id: self.config.chain_id.clone(),
+                                    hash: *self.chain.genesis().hash(),
+                                },
+                                height: self.get_height(&head),
+                                tracked_shards,
```

Comment:
Is there a plan to deprecate `tracked_shards`?

Reply:
Yes, I just probably don't have time to do it before sharding launch since I still have a lot other to finish. 

The plan is to make neard automatically update the client config so that the validators won't need to worry about it at all

Reply:
https://github.com/near/nearcore/issues/4930